### PR TITLE
Issue#90/input review container

### PR DESCRIPTION
**Pull request by arya56**
# Description
Create Component to add Review editor
Please provide a short summary explaining what this PR is about.
should have this design
image

Fixes # (issue)
#90
# How to test?
see in story book
Please provide a short summary how your changes can be tested?

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        <textarea value={inputValue} placeholder={placeHolderText} onChange={onChange}/>
+      </div>
+      <div className="button-part">
+        <Button
+          buttonName="Send review"
+          style={buttonStyle}
+          type="submit"
+          onClick={() => {
+            ('clicked');
+          }}
```
Comment (arya56) [inline src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:44]: @pdsorensen Since the task/issue hasn't been cleared here I need the suggestion to make this onClick functional.

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    setInputValue((event.target.value))
+  };
+  return (
+    <div className="container">
+      <div className="top-part">
+        <div className="profile-picture">
+          <div className="circle-mask">
+            <img
+              src={avatarUrl || unknownUser}
+              alt="Portrait of user"
```
Comment (arya56) [inline src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:25]: @pdsorensen For some reason the "unknownUser" doesn't show the picture! It suppose to bring a fallback pic in case user doesn't have one. Can be seen in the second scenario of Storybook.

Comment (pdsorensen) [reply src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:25]: Your image import should be `import unknownUser from '../../assets/images/portraits/unknown-black.jpg';`

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.js (hunk 1 above)
Comment (pdsorensen) [reply src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:44]: I don't believe the backend is ready? Maybe ask in slack and ask whether the backend for adding reviews has been implemented. 

If not, just leave the function with an empty body.

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.stories.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+// eslint-disable-next-line import/no-named-as-default
```
Comment (pdsorensen) [inline src/client/components/ReviewEditorComponent/ReviewEditorComponent.stories.js:2]: Why is this needed?

**Review by pdsorensen — CHANGES_REQUESTED** [review]
Please fix the github actions. :)

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  avatarUrl,
+  fullName,
+  inputText,
+  placeHolderText,
+  buttonStyle,
+}) {
+  const [inputValue,setInputValue] = useState()
+  const onChange = (event) => {
+    setInputValue((event.target.value))
+  };
```
Comment (arya56) [inline src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:17]: I guess we need to lifting up the "inputValue" state  or? @pdsorensen

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.stories.js (hunk 3 above)
Comment (arya56) [reply src/client/components/ReviewEditorComponent/ReviewEditorComponent.stories.js:2]: removed

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.js (hunk 2 above)
Comment (pdsorensen) [reply src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:25]: Sorry, edited it to be correct. It should be png.

Comment (arya56) [reply src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:25]: Unfortunately no succeed, have checked beforehand!

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.js (hunk 4 above)
Comment (arya56) [reply src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:17]: should add functionality in this version?

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.js (hunk 2 above)
Comment (pdsorensen) [reply src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:25]: Keep debugging @arya56. I believe you can do it. :)

**Comment by pdsorensen** [comment]
You still have github steps failing, please fix them before adding me as a reviewer. 

image

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        />
+      </div>
+      <div className="button-part">
+        <Button
+          buttonName="Send review"
+          style={buttonStyle}
+          type="submit"
+          onClick={() => {
+            // ('clicked');
+            returnTextState(inputValue)
```
Comment (arya56) [inline src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:48]: @pdsorensen is it the correct way to return the inputValue?

Comment (pdsorensen) [reply src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:48]:
```suggestion
            onClick={() => { console.log("Waiting for backend to expose endpoint for reviewing"})
```

Comment (arya56) [reply src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:48]: @pdsorensen  in this case shall remove line 12 as well?

Comment (pdsorensen) [reply src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:48]: Hm, why is it relevant?

Comment (arya56) [reply src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:48]: @pdsorensen have confusion, assumed that the "returnTextState" is a callback function to return the "inputValue" text to the parent component. right?

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.js (hunk 2 above)
Comment (arya56) [reply src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:25]: @pdsorensen still no success here.

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+}
+ReviewEditorComponent.propTypes = {
+  avatarUrl: PropTypes.string,
+  fullName: PropTypes.string.isRequired,
+  returnTextState: PropTypes.func,
+  placeHolderText: PropTypes.string,
+  buttonStyle: PropTypes.string,
+};
+ReviewEditorComponent.defaultProps = {
+  avatarUrl: {},
```
Comment (valmdz) [inline src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:63]:
```suggestion
  avatarUrl: '',
```

Comment (valmdz) [reply src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:63]: That fixes the issue with the unkownUser not showing up ;)

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  returnTextState: {},
+  placeHolderText: {},
+  buttonStyle:{ backgroundColor: 'blue', fontWeight: 'bold' }
```
Comment (valmdz) [inline src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:66]:
```suggestion
  buttonStyle:{ backgroundColor: '#0676F2', fontWeight: 'bold' }
```

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.style.css [hunk 8]
```diff
@@ -0,0 +1,66 @@
+.container {
+  max-width: 598px;
```
Comment (valmdz) [inline src/client/components/ReviewEditorComponent/ReviewEditorComponent.style.css:2]:
```suggestion
  width: 100%;
```

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.style.css [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+  align-items: center;
+  display: flex;
+  flex-direction: row;
+}
+.input-part textarea {
+  border-radius: 5px;
+  border: 1px solid #9a9a9a;
+  align-items: flex-end;
+  height: 120px;
+  min-width: 550px;
```
Comment (valmdz) [inline src/client/components/ReviewEditorComponent/ReviewEditorComponent.style.css:60]: add `width: 100%;`

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.style.css (hunk 8 above)
Comment (valmdz) [reply src/client/components/ReviewEditorComponent/ReviewEditorComponent.style.css:2]: To help and make it more responsive, given that we don't know yet how much space will be taken by the other elements in the same page where this will be use

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.style.css [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+.user-info {
+  align-items: flex-start;
+}
+.user-name {
+  font-weight: bold;
+  font-size: medium;
+  overflow: hidden;
+  min-height: 1em;
+  max-height: 2.5em;
+}
```
Comment (valmdz) [inline src/client/components/ReviewEditorComponent/ReviewEditorComponent.style.css:48]: Just keep:
```
.user-name {
  font-weight: bold;
  font-size: medium;
}
```

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.js [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+        <div className="profile-picture">
+          <div className="circle-mask">
+            <img
+              src={avatarUrl || unknownUser}
+              alt="Portrait of user"
+              className="user-photo"
+            />
+          </div>
+        </div>
+        <div className="user-info">
```
Comment (valmdz) [inline src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:30]:
```suggestion
        
```

Comment (valmdz) [reply src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:30]: See more below in the Css part so this makes more sense ;)

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.js [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+          <div className="user-name">{fullName}</div>
+        </div>
```
Comment (valmdz) [inline src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:32]:
```suggestion
        
```

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.style.css [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+.user-photo {
+  display: inline;
+  margin: 0 auto;
+  width: 40px;
+  justify-content: flex-start;
+}
+.user-info {
+  align-items: flex-start;
+}
```
Comment (valmdz) [inline src/client/components/ReviewEditorComponent/ReviewEditorComponent.style.css:41]: Delete:
```.user-info {
  align-items: flex-start;
}
```

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.style.css (hunk 10 above)
Comment (valmdz) [reply src/client/components/ReviewEditorComponent/ReviewEditorComponent.style.css:48]: I think in your case this makes more sense because your component has more room than mine ;)

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.js [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+  returnTextState
+}) {
+  const [inputValue, setInputValue] = useState();
+  const onChange = (event) => {
+    setInputValue(event.target.value);
+  };
+  return (
+    <div className="container">
+      <div className="top-part">
+        <div className="profile-picture">
```
Comment (valmdz) [inline src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:21]: When you bring in the stars you might need to add a `div `that contains the `profile-picture` and the `user-name`, and keep the stars out. Then with flex you can` justify-content: space-between` to keep the container with the image and the user name on one side and the stars on the other side.

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.js (hunk 11 above)
Comment (valmdz) [inline src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:30]:
```suggestion
        
```

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.js (hunk 12 above)
Comment (valmdz) [inline src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:32]:
```suggestion
       
```

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.js (hunk 11 above)
Comment (valmdz) [reply src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:30]: See below in the CSS so it makes more sense ;)

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.style.css (hunk 13 above)
Comment (valmdz) [inline src/client/components/ReviewEditorComponent/ReviewEditorComponent.style.css:41]: delete
```
.user-info {
  align-items: flex-start;
}
```

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.style.css (hunk 10 above)
Comment (valmdz) [inline src/client/components/ReviewEditorComponent/ReviewEditorComponent.style.css:48]: Just keep
```
.user-name {
  align-items: flex-start;
  font-weight: bold;
  font-size: medium;
}
```

Comment (valmdz) [reply src/client/components/ReviewEditorComponent/ReviewEditorComponent.style.css:48]: In your case, since your component has more room and probably the names will not be soooo long to overflow the component, you might be ok with my suggestion

**Comment by pdsorensen** [comment]
@arya56 are you still working on this? was is possible to implement the suggestions from @valmdz?

**Comment by arya56** [comment]
> @arya56 are you still working on this? was is possible to implement the suggestions from @valmdz?

@pdsorensen will finalize it soon.

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.js (hunk 14 above)
Comment (arya56) [reply src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:21]: Great. will do it. :+1:

File: src/client/components/ReviewEditorComponent/ReviewEditorComponent.js (hunk 2 above)
Comment (arya56) [reply src/client/components/ReviewEditorComponent/ReviewEditorComponent.js:25]: solved, with last update loader dependency.

**Review by arya56 — COMMENTED** [review]
@pdsorensen  Reviews added. Thanks you both. @valmdz

File: src/client/components/ReviewPost/ReviewEditorComponent/ReviewEditorComponent.js [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+      </div>
+    </div>
+  );
+}
+
+ReviewEditorComponent.propTypes = {
+  avatarUrl: PropTypes.string,
+  fullName: PropTypes.string.isRequired,
+  placeHolderText: PropTypes.string,
+  buttonStyle: PropTypes.string,
```
Comment (arya56) [inline src/client/components/ReviewPost/ReviewEditorComponent/ReviewEditorComponent.js:62]: @pdsorensen  To resolve the following error, after changed ```buttonStyle``` prototype to object, and the related modification in Button component problem solved, but Eslint show violatio: ```prop type object is forbidden``` any suggestion? (I will not push changes related to Button) :smile_cat: 

image

**Review by arya56 — COMMENTED** [review]
Here I entitled the prop type error in the cosole.

File: src/client/components/ReviewPost/ReviewEditorComponent/ReviewEditorComponent.js (hunk 15 above)
Comment (pdsorensen) [reply src/client/components/ReviewPost/ReviewEditorComponent/ReviewEditorComponent.js:62]: Can't we remove button styles as a prop argument, and define them in the `ReviewEditorComponent`?

Comment (arya56) [reply src/client/components/ReviewPost/ReviewEditorComponent/ReviewEditorComponent.js]: @pdsorensen Done, but still have warning, I think this related to Button proptypes, does it? I think we can ignore it for now and finalize this, and try to solve the Button prop problem in another story.
image

**Review by pdsorensen — APPROVED** [review]
🙌
